packages/lodestar/src/chain/blocks/process.ts

```diff
@@ -126,27 +131,43 @@ export function updateForkChoice(
 }
 
 export async function runStateTransition(
-  config: IBeaconConfig,
-  db: IBeaconDb,
-  logger: ILogger,
-  stateContext: Required<IStateContext>,
+  emitter: ChainEventEmitter,
+  stateContext: ITreeStateContext,
   job: IBlockProcessJob
 ): Promise<IStateContext | null> {
+  const config = stateContext.epochCtx.config;
   try {
+    const preSlot = stateContext.state.slot;
+    const postSlot = job.signedBlock.message.slot;
+    const preEpoch = computeEpochAtSlot(config, preSlot);
+    let nextEpochSlot = computeStartSlotAtEpoch(config, preEpoch + 1);
+    while (nextEpochSlot < postSlot) {
+      processSlots(stateContext.epochCtx, stateContext.state, nextEpochSlot);
+      const checkpoint: Checkpoint = {
+        root: stateContext.state.blockRoots[(nextEpochSlot - 1) % config.params.SLOTS_PER_HISTORICAL_ROOT],
```

Comment:
this results in `getBlockRootAtSlot(nextEpochSlot - 1)`. Did you mean `getBlockRootAtSlot(nextEpochSlot )`?

Reply:
I think we need something like:
```typescript
const blockHeader = stateContext.state.latestBlockHeader.clone();
blockHeader.stateRoot = stateContext.state.hashTreeRoot();
const root = blockHeader.hashTreeRoot();
```

Below, we can just hashTreeRoot the job.block.message